grpc/stream_test.go

```diff
@@ -0,0 +1,264 @@
+				new grpc.Stream(client, "grpc.testing.TestService/EmptyCall", { timeout: true })`,
+				err: "invalid timeout value: unable to use type bool as a duration value",
+			},
+		},
+		{
+			name: "InvalidTimeout",
+			initString: codeBlock{code: `
+				var client = new grpc.Client();
+				client.load([], "../vendor/google.golang.org/grpc/test/grpc_testing/test.proto");`},
+			vuString: codeBlock{
+				code: `
+				client.connect("GRPCBIN_ADDR");
+				new grpc.Stream(client, "grpc.testing.TestService/EmptyCall", { timeout: "please" })`,
+				err: "invalid duration",
+			},
+		},
+		{
+			name: "StringTimeout",
+			initString: codeBlock{code: `
+				var client = new grpc.Client();
+				client.load([], "../vendor/google.golang.org/grpc/test/grpc_testing/test.proto");`},
+			vuString: codeBlock{
+				code: `
+				client.connect("GRPCBIN_ADDR");
+				new grpc.Stream(client, "grpc.testing.TestService/EmptyCall", { timeout: "1h42m" })`,
+			},
+		},
+		{
+			name: "FloatTimeout",
+			initString: codeBlock{code: `
+				var client = new grpc.Client();
+				client.load([], "../vendor/google.golang.org/grpc/test/grpc_testing/test.proto");`},
+			vuString: codeBlock{
+				code: `
+				client.connect("GRPCBIN_ADDR");
+				new grpc.Stream(client, "grpc.testing.TestService/EmptyCall", { timeout: 400.50 })`,
+			},
+		},
+		{
+			name: "IntegerTimeout",
+			initString: codeBlock{
+				code: `
+				var client = new grpc.Client();
+				client.load([], "../vendor/google.golang.org/grpc/test/grpc_testing/test.proto");`,
+			},
+			vuString: codeBlock{
+				code: `
+				client.connect("GRPCBIN_ADDR");
+				new grpc.Stream(client, "grpc.testing.TestService/EmptyCall", { timeout: 2000 })`,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			ts := setup(t)
+
+			m, ok := New().NewModuleInstance(ts.VU).(*ModuleInstance)
+			require.True(t, ok)
+			require.NoError(t, ts.VU.Runtime().Set("grpc", m.Exports().Named))
+
+			// setup necessary environment if needed by a test
+			if tt.setup != nil {
+				tt.setup(ts.httpBin)
+			}
+
+			replace := func(code string) (goja.Value, error) {
+				return ts.VU.Runtime().RunString(ts.httpBin.Replacer.Replace(code))
+			}
+
+			val, err := replace(tt.initString.code)
+			assertResponse(t, tt.initString, err, val, ts)
+
+			registry := metrics.NewRegistry()
+			root, err := lib.NewGroup("", nil)
+			require.NoError(t, err)
+
+			state := &lib.State{
+				Group:     root,
+				Logger:    ts.logger,
+				Dialer:    ts.httpBin.Dialer,
+				TLSConfig: ts.httpBin.TLSClientConfig,
+				Samples:   ts.samples,
+				Options: lib.Options{
+					SystemTags: metrics.NewSystemTagSet(
+						metrics.TagName,
+						metrics.TagURL,
+					),
+					UserAgent: null.StringFrom("k6-test"),
+				},
+				BuiltinMetrics: metrics.RegisterBuiltinMetrics(registry),
+				Tags:           lib.NewVUStateTags(registry.RootTagSet()),
+			}
+			ts.MoveToVUContext(state)
```

Comment:
We could have a MoveToVUContext method on the test state setting the default State, and overwriting the inherited so we don't need to replicate this logic across the tests. If we need to change a specific field we still should be able to do it after.

```
ts.MoveToVUContext() // or ts.InitState()
ts.StateField.Group = "my-single-changed-field"
````